### DISCUSSION TITLE
PressetAssets and Dynamic Assets

### DISCUSSION THREAD

**Discussion by AbbeyDesign** [solspace/craft-freeform#650] [Feature Requests]
I have a form that an admin fills out. The notification goes to a user with an attachment. 

The form has a hidden field that gets the id of this existing craft asset. I would like for the notification to send with the asset attached so in my notification file my code looks like this:

```
{# includeAttachments: true #}
{# presetAssets: [where my problem is] #}
```
My issue is it works as intended if I hard code an ID into the notification template, eg '12345' but it does not work if I try to do it dynamically at the template level.

This works
```
{# presetAssets: 12345 #}
```

These do not work:
```
{# presetAssets: {{ fieldhandle }} #}
{# presetAssets: {{ fieldhandle.value }} #}
{# presetAssets: {{ submission.fieldhandle }} #}
{# presetAssets: {{ submission.fieldhandle.value }} #}
```

I have used this approach on some other lines on notification templates... eg doing this with the "subject" works as intended:
```
{# subject: {{ fieldhandle }} #}
```

**Comment by AbbeyDesign** [comment]
Rereading your docs, the features listed under "PressetAssets" do not include this line as others do:

"May include any Freeform field variable (fieldName)."

So it's not possible? Could there be any workaround?

My use case is the admin must send out pdf invoices as attachments, not links.

**Comment by kjmartens** [comment]
Hi @AbbeyDesign,

Thanks for your feedback on this. As you have realized, this is not currently possible with Freeform. However, I checked with a developer and it might be something that we can add support for soon. Which version of Freeform are you on? 🙂

**Reply by AbbeyDesign** [reply]
Craft 4.4.5 and Freeform 4.0.21

I poked around in the plugin. I was trying to find a line or two I could change to at least bypass the code where 'presetAssets' is not replacing this part: "{{ whatIwanttochange }}" and acted more like the other fields as a replaceable string like th fromName, fromEmail, subject etc.

I only need one attachment so it does not even have to be an array. But as previously stated it works if a number is in there. The template actually breaks in the cp because craft says the first item "{{" in the array is not a number.

 I just need a way to replace that number on a per entry basis.

**Reply by kjmartens** [reply]
We will be checking into adding support for this next week. The tricky part is accounting for both an element picker in the control panel and the possibility of twig grabbing something dynamically in there. 🙂

**Reply by AbbeyDesign** [reply]
Great, thanks!

**Comment by kjmartens** [comment]
We've now added support for this in Freeform 4.0.23. 🙂
